src/main/java/com/northcoders/makemydayapi/dto/AuthRequestDTO.java

```diff
@@ -0,0 +1,32 @@
+package com.northcoders.makemydayapi.dto;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.Getter;
+
+
+@Getter
+@Data
+@AllArgsConstructor
+public class AuthRequestDTO {
+
+    @NotBlank(message ="First name is required")
+    @Size(min = 1)
+    private String firstname;
+
+    @NotBlank(message = "Last name is required")
+    @Size(min = 1)
+    private String lastname;
+
+    @NotBlank(message = "Email is required")
+    @Email(message = "Email must be valid")

```

Comment:
- We can also [pass our custom regex](https://stackoverflow.com/a/65371615/16855913) into this `@Email` annotation, to solve validation problem on emails.
- We can also create our `@ValidEmail` annotation that validates emails with our custom regex.

So, it can be refactored later.

Reply:
I didn't know how to test the annotation

Reply:
I used them but I couldn't find a way to test them which is why I used manual validation, I think it had something to do with the fact we were using mocks

Reply:
In advance, testing the annotation can be accomplished in a separate `dto` test, **a test file** for a request body dto that we **validate**:

- **Do whatever Jackson does behind the scenes:** Manually create a dto object instance with valid/invalid email values, so annotation validation is triggered there during dto object initialisation.

But, I think we can trust our regex for now, without any tests. We can deal with it later.

Reply:
Okay sounds good thanks!